ios/TPSStripe/TPSStripeManager.m

```diff
@@ -152,14 +305,252 @@ - (NSDictionary *)constantsToExport
+                                                                                                NSDictionary *jsError = [errorCodes valueForKey:kErrorKeyApi];
+                                                                                                [self rejectPromiseWithCode:jsError[kErrorKeyCode] message:error.localizedDescription];
+                                                                                                return;
+                                                                                            }
+
+                                                                                            switch (status) {
+                                                                                                case STPPaymentHandlerActionStatusSucceeded:
+                                                                                                case STPPaymentHandlerActionStatusCanceled:
+                                                                                                    // Succeeded/canceled should all be attached to the intent
+                                                                                                    [self resolvePromise: [self convertConfirmPaymentIntentResult: intent]];
+                                                                                                    break;
+                                                                                                case STPPaymentHandlerActionStatusFailed:
+                                                                                                    // This should not happen, as we should respond with an error -- what should we do?
+                                                                                                    [self rejectPromiseWithCode:[errorCodes valueForKey:kErrorKeyApi][kErrorKeyCode] message:@"FAILED"];
+                                                                                                    break;
+                                                                                            }
+                                                                                        }];
+                                 } else {
+                                     // We can't do anything else for the other intent status cases, so let's just return control to the App
+                                     requestIsCompleted = YES;
+                                     [self resolvePromise: [self convertConfirmPaymentIntentResult:intent]];
+                                 }
+                             }];
+}
+
+RCT_EXPORT_METHOD(authenticatePayment:(NSDictionary<TPSStripeType(authenticatePayment), id>*)params
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+
+    NSString * clientSecret = TPSStripeParam(authenticatePayment, clientSecret);
```

Comment:
Should this be `NSString * clientSecret = params[TPSStripeParam(confirmPayment, clientSecret)];`

Reply:
You’re absolutely right, thanks!
```suggestion
    NSString * clientSecret = params[TPSStripeParam(confirmPayment, clientSecret)];
```